x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/ShrinkAction.java

```diff
@@ -278,40 +299,52 @@ public List<Step> toSteps(Client client, String phase, Step.StepKey nextStepKey)
             ShrinkIndexNameSupplier::getShrinkIndexName
         );
         DeleteStep deleteSourceIndexStep = new DeleteStep(deleteIndexKey, isShrunkIndexKey, client);
-        ShrunkenIndexCheckStep waitOnShrinkTakeover = new ShrunkenIndexCheckStep(isShrunkIndexKey, nextStepKey);
-        return Arrays.asList(
-            conditionalSkipShrinkStep,
-            checkNotWriteIndexStep,
-            waitForNoFollowersStep,
-            waitUntilTimeSeriesEndTimeStep,
-            readOnlyStep,
-            checkTargetShardsCountStep,
-            cleanupShrinkIndexStep,
-            generateUniqueIndexNameStep,
-            setSingleNodeStep,
-            checkShrinkReadyStep,
-            shrink,
-            allocated,
-            copyMetadata,
-            isDataStreamBranchingStep,
-            aliasSwapAndDelete,
-            waitOnShrinkTakeover,
-            replaceDataStreamBackingIndex,
-            deleteSourceIndexStep
+        ShrunkenIndexCheckStep waitOnShrinkTakeover = new ShrunkenIndexCheckStep(isShrunkIndexKey, lastOrNextStep);
+
+        List<Step> steps = new ArrayList<>(
+            List.of(
```

Comment:
This looks admittedly a bit wonky. It seemed the best way way to make a mutable literal list. I played with making `steps` immutable then copying it to add the `allowWriteAfterShrinkStep`, but didn't like that much either. :man_shrugging:

Reply:
You can also use a stream of steps, use null if the step shouldn't be there and then filter out nulls, amd convert it to list. We use it in certain places like cluster settings.

Reply:
Nice, I like it!